src/handler.rs

```diff
@@ -577,9 +609,239 @@ where
 
         Ok(())
     }
+
+    fn get_inflight_fd(
+        &mut self,
+        inflight: &vhost::vhost_user::message::VhostUserInflight,
+    ) -> VhostUserResult<(vhost::vhost_user::message::VhostUserInflight, File)> {
+        let ret_val = -1;
+        // Total size of the inflight queue region
+        let mut total_mmap_size =
+            self.get_inflight_queue_size(inflight.queue_size) * inflight.num_queues as usize;
+
+        // Create a memfd region to hold the queues for inflight I/O tracking
+        let (mmap_ptr, inflight_fd, dup_inflight_fd) = self.memfd_alloc(total_mmap_size)?;
```

Comment:
Seems it will get simpler if moves following "sef.inflight_state.set_inflight_state()" into memfd_alloc()